### PR TITLE
sql/mysql: support MarshalSpec

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: sql/mysql/sqlspec.go [hunk 1]
```diff
@@ -34,6 +38,26 @@ func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{
 	return fmt.Errorf("mysql: failed unmarshaling spec. %T is not supported", v)
 }
 
+// MarshalSpec marshals v into an Atlas DDL document using a schemaspec.Marshaler.
```
Comment (yonidavidson) [inline sql/mysql/sqlspec.go:41]: is v a known name for interface?

Comment (rotemtam) [reply sql/mysql/sqlspec.go:41]: https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/encoding/json/encode.go;l=158

**Comment by yonidavidson** [comment]
I meant in the go doc.

File: sql/mysql/sqlspec.go [hunk 2]
```diff
@@ -11,12 +13,14 @@ import (
 	"ariga.io/atlas/sql/sqlspec"
 )
 
+type ddlDoc struct {
```
Comment (a8m) [inline sql/mysql/sqlspec.go:16]: why not just `doc`?

File: sql/mysql/sqlspec.go [hunk 3]
```diff
@@ -34,6 +38,26 @@ func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{
...
+	}
+	spec, tables, err := schemaSpec(s)
+	if err != nil {
+		return nil, fmt.Errorf("mysql: failed converting schema to spec: %w", err)
+	}
+	doc := &ddlDoc{
+		Tables:  tables,
+		Schemas: []*sqlspec.Schema{spec},
+	}
+	return marshaler.MarshalSpec(doc)
```
Comment (a8m) [inline sql/mysql/sqlspec.go:58]:
```suggestion
	return marshaler.MarshalSpec(doc{
		Tables:  tables,
		Schemas: []*sqlspec.Schema{spec},
	})
```

**Pull request merged**